python/ray/serve/_private/replica.py

```diff
@@ -488,15 +489,29 @@ def user_health_check():
 
         self.restart_counter.inc()
 
+        self.metrics_pusher = self.metrics_pusher = MetricsPusher()
         if autoscaling_config:
             process_remote_func = controller_handle.record_autoscaling_metrics.remote
             config = autoscaling_config
-            self.metrics_pusher = MetricsPusher(
-                process_remote_func,
-                config.metrics_interval_s,
+            self.metrics_pusher.register_task(
                 self._collect_autoscaling_metrics,
+                config.metrics_interval_s,
+                process_remote_func,
             )
-            self.metrics_pusher.start()
+
+        self.metrics_pusher.register_task(
+            self._set_replica_requests_metrics,
+            RAY_SERVE_GAUGE_METRIC_SET_PERIOD_S,
+        )
+        self.metrics_pusher.start()
+
+    def _set_replica_requests_metrics(self):
+        request_stats = self._get_handle_request_stats()
+        if request_stats:
+            num_running_requests = request_stats["running"]
+            num_pending_requests = request_stats["pending"]
+            self.num_processing_items.set(num_running_requests)
+            self.num_pending_items.set(num_pending_requests)
```

Comment:
it's kind of confusing that we separate these here but for autoscaling they're combined :'(